src/rpc/rawtransaction_util.cpp

```diff
@@ -545,8 +545,9 @@ bool ValidateTransactionPeginInputs(const CMutableTransaction& mtx, std::map<int
             continue;
         }
         // Report warning about immature peg-in though
-        if(txin.m_is_pegin && !IsValidPeginWitness(mtx.witness.vtxinwit[i].m_pegin_witness, fedpegscripts, txin.prevout, err, true)) {
-            CHECK_NONFATAL(err == "Needs more confirmations.");
+        bool depth_failed = false;
+        if(txin.m_is_pegin && !IsValidPeginWitness(mtx.witness.vtxinwit[i].m_pegin_witness, fedpegscripts, txin.prevout, err, true, &depth_failed)) {
+            assert(depth_failed);
```

Comment:
This check (the string error version) changed from `assert` to `CHECK_NONFATAL` since 0.18, apparently. I can't tell whether you changed it back to assert intentionally when rebasing my changes.

Reply:
I did consciously change this to an `assert`, though I didn't think much about the code history. Let me change it back because I don't remember my reasoning.